transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -365,7 +383,11 @@ public ChannelFuture writeOneInbound(Object msg) {
      */
     public ChannelFuture writeOneInbound(Object msg, ChannelPromise promise) {
         if (checkOpen(true)) {
-            pipeline().fireChannelRead(msg);
+            if (!config().isAutoRead()) {
+                tempInboundMessages().add(msg);
```

Comment:
imho this is not correct.. shouldn't we store the future together with the msg and only notify it once we dispatched the msg through the pipeline ?